src/devices/HuskyLens/HuskyLens.cs

```
@@ -0,0 +1,149 @@
+                command[7] = command.CalculateChecksum();
+                _connection.Write(command);
+                WaitForOK();
+
+                _algorithm = value;
+            }
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="connection">guess</param>
+        public HuskyLens(IBinaryConnection connection)
+        {
+            _connection = connection;
+            Algorithm = Algorithm.Undefined;
+        }
+
+        /// <summary>
+        /// Test connection with HuskyLens.
+        /// </summary>
+        /// <returns>true for success, guess the rest</returns>
+        public bool Ping()
+        {
+            // COMMAND_REQUEST_KNOCK(0x2C):
+            // Used for test connection with HUSKYLENS.When HUSKYLENS received this command, HUSKYLENS will return COMMAND_RETURN_OK
+            // See: https://github.com/HuskyLens/HUSKYLENSArduino/blob/master/HUSKYLENS%20Protocol.md#command_request_knock0x2c
+            _connection.Write(new byte[] { 0x55, 0xAA, 0x11, 0x00, 0x2C, 0x3C });
+            WaitForOK();
+            return true;
+        }
+
+        /// <summary>
+        /// Gets all recognized objects
+        /// </summary>
+        public IReadOnlyCollection<HuskyObject> GetAllObjects()
+        {
+            // COMMAND_REQUEST (0x20):
+            // Request all blocks and arrows from HUSKYLENS.
+            // See: https://github.com/HuskyLens/HUSKYLENSArduino/blob/master/HUSKYLENS%20Protocol.md#command_request-0x20
+            var command = new byte[] { 0x55, 0xAA, 0x11, 0x00, 0x20, 0x30 };
+            _connection.Write(command);
+
+            // wait for COMMAND_RETURN_INFO
+            var response = new DataFrame(_connection.Read(0x0A + 6));
+            if (!response.IsValid() || response.Command != Command.COMMAND_RETURN_INFO)
+            {
+                // error
+                throw new NotImplementedException();
```

if this is an error please throw InvalidOperationException, otherwise please add comment why this is not implemented and what needs to happen here exactly